toolchain/check/check.cpp

```diff
@@ -63,11 +66,14 @@ class SemIRDiagnosticConverter : public DiagnosticConverter<SemIRLocation> {
       }
     };
 
-    // Parse nodes always refer to the current IR.
-    if (!loc.is_inst_id) {
+    if (loc.is_inst_id) {
+      cursor_inst_id = loc.inst_id;
+    } else {
+      // Parse nodes always refer to the current IR.
```

Comment:
If so, do you need to use `handle_loc` here, or could you call `ConvertLocationInFile` directly?

I ask because omitting the `handle_loc` call here would enable a couple other changes:
- It looks like `handle_loc` is only called in one other place, and I think it would be clearer to inline it there, avoiding the indirection of a function call.
- I also think it would be clearer if this block were above the lambdas, since it partly deals with the initialization of `cursor_inst_id`, and the lambdas seem more related to the body of the loop below.

Reply:
Sorry, moved this comment to a more appropriate spot.